src/compiler/checker.ts

```diff
@@ -5118,7 +5119,8 @@ namespace ts {
         function getSignaturesOfSymbol(symbol: Symbol): Signature[] {
             if (!symbol) return emptyArray;
             const result: Signature[] = [];
-            for (let i = 0, len = symbol.declarations.length; i < len; i++) {
+            const len = symbol.declarations.length;
+            for (let i = 0; i < len; i++) {

```

Comment:
Inner lines of code uses `i` for some purpose. It seems I can still switch to for-of in this case but should I do it?